TACC/Core-Portal
# WP-553: Search index cron tab fixes

## Overview
Search index for community and public data is built on a cron schedule in the background. 
To use the schedule and trigger tasks `Celery beat` needs to be run. Celery beat is responsible for triggering periodic tasks defined in your Django application. More on Celery beat is [here](https://github.com/celery/django-celery-beat).

Currently celery beat is not run in core portal based images.

## Related

* [WP-553](https://tacc-main.atlassian.net/browse/WP-553)

## Changes
1. Add celery beat to celery portal worker. Note: an additional change will happen in camino also. This PR only fixes local changes.
2. Once the celery task is run, the traversal of files has a bug, this PR fixes it.


## Testing

1.  Add this line to `server/portal/settings/settings_local.py` (adjust the minute accordingly to your schedule). 15 here says run at 15th minute in every hour.
```
COMMUNITY_INDEX_SCHEDULE = {'hour': '*', 'minute': 15}
```
2. Start server (make build and make start)
3. Setup core portal search. See steps [here](https://github.com/TACC/Core-Portal?tab=readme-ov-file#setting-up-search-index)
4. Check your search indexes. Run this in a browser: `http://127.0.0.1:9200/_cat/indices?v`
5. From previous step, check the index that starts with `cep-dev-files` and check that docs.count is 0. If not, clear the index using command below: (instead of cep-dev-files-2024_04_30_18_39_43_758464, replace it with the index name of your file index).
```
curl -XPOST "http://127.0.0.1:9200/cep-dev-files-2024_04_30_18_39_43_758464/_delete_by_query" -H 'Content-Type: application/json' -d'
{
  "query": {
    "match_all": {}
  }
}
'
```
6. Wait for 15th minute of the hour.
7. Check logs for info like
```
INDEXING Community Data SYSTEM
INDEXING Public Data SYSTEM
```
8. Check your search indexes again. Run this in a browser: `http://127.0.0.1:9200/_cat/indices?v`  The files index should have docs.count much higher than before. Definitely not 0.

***Artifacts from my testing***

Search index data after background index :
```
health status index                                          uuid                   pri rep docs.count docs.deleted store.size pri.store.size
green  open   .geoip_databases                               VRzw89swRRuDi2moL6B-XQ   1   0         33           33     31.1mb         31.1mb
yellow open   cep-dev-allocations-2023_11_06_20_02_18_037213 imfQ33QYQbKUVZ-_RG1bZw   1   1          1            0       48kb           48kb
yellow open   cep-dev-projects-2023_11_06_20_02_14_375336    757qrYjXTlqQOkvbEtgG3A   1   1         16            0     14.9kb         14.9kb
yellow open   cep-dev-files-2024_04_30_18_39_43_758464       1KMfmGoiTkyGLbbMYoGmug   3   1         62           46    251.3kb        251.3kb
yellow open   cep-dev-cms                                    arhSYrnfQFGGog3uVas48A   1   1          1            0      7.8kb          7.8kb

```

Logs:
```
[2024-04-30 19:08:00,036: INFO/ForkPoolWorker-2] INDEXING Community Data SYSTEM
[2024-04-30 19:08:00,048: INFO/ForkPoolWorker-2] INDEXING Public Data SYSTEM
```

## UI



## Notes
Camino changes are needed to match docker compose file in this PR.
